### DISCUSSION TITLE
Is resampling a stream possible?

### DISCUSSION THREAD

**Discussion by gadgetbazza** [blakeblackshear/frigate#14589] [Ask A Question] [answered]
Hi,

Fairly long time user and sponsor of Frigate here, love the work you guys have been doing....

So I recently added a new fisheye camera from Reolink (FE-P).  The jury is still out on how much I like it, but after testing a number of the views, the one that seems to provide me with the best coverage has an area of the image that isn't necessary.  I'm probably not explaining myself properly, so I'm attaching an image of the output.  It's also worth mentioning that the standard image output is of a square format rather than the typical rectangular feed.

What I'm looking to do ideally is a way to resample this feed to show only the area that isn't black in the attached image.  Essentially creating a new feed that is closer to a rectangular image and provides better visibility of the useful part of the image in the Frigate home page and also in my Home Assistant cards.  Kind of like a motion mask, but applied to visible stream as a whole.  Is there a way to do this within Frigate or any other solutions you can think of?

Thanks
Barry

![Screenshot 2024-10-25 at 13 43 25](https://github.com/user-attachments/assets/ba3e9cbd-b0aa-48d4-a648-741d462299bd)

**Comment by hawkeye217** [comment, marked as answer]
It's possible, but it will likely be CPU intensive. You'll want to use go2rtc with the ffmpeg source and emply some cropping filters.

Some users have done similar work here, although it was for dewarping a fisheye video. The same principles can be used for cropping.

https://github.com/blakeblackshear/frigate/discussions/13326

**Reply by gadgetbazza** [reply]
Thanks Hawkeye, don't know why I didn't think to search for the word "crop" rather than "resample"!

Anyway, got it re-sampled now, after a bit of confusion where the Config Editor was lower casing my RTC stream names in the camera sections and it giving a 404 error on that stream accordingly.

Here's my camera config for the fisheye camera and the resulting output.

Thanks!

Barry

`
go2rtc:
  webrtc:
    candidates:
      - x.x.x.x:8555
      - stun:8555
  streams:
    Lounge-Internal:
      - ffmpeg:rtsp://admin:kak1m0t0@192.168.4.23:554/Preview_01_main#video=h265#audio=copy#raw=-vf crop=w=3200:h=2200:x=0:y=0
    Lounge-Internal-Sub:
      - ffmpeg:rtsp://admin:kak1m0t0@192.168.4.23:554/Preview_01_sub#video=h265#raw=-vf crop=w=1024:h=780:x=0:y=0

cameras:
  Lounge-Internal:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/Lounge-Internal-Sub
          input_args:
            - -rtsp_transport
            - tcp
            - -timeout
            - '9000000'
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/Lounge-Internal
          input_args:
            - -rtsp_transport
            - tcp
            - -timeout
            - '9000000'
          roles:
            - record
            - audio
    detect:
      width: 1024
      height: 800
    objects:
      track:
        - person
        - dog
   `

and the resulting image:

![Screenshot 2024-10-25 at 17 10 36](https://github.com/user-attachments/assets/37811d86-53f4-473d-bd2e-a781282bf772)

**Reply by hawkeye217** [reply]
Looks great! Good work.
